Add local registry support for remote builders

djmb opened the pull request:
When running with a remote builder and a local registry:
1. Forward the local registry port to the remote builder
2. Use host networking on the remote builder
3. Use a different builder name as the settings are different

A remote builder can only have one port at a time forwarded, so concurrent builds on the same registry port will fail.

See: https://github.com/basecamp/kamal/issues/1674#issuecomment-3424910818

cc @jeromedalbert, @adrienpoly

darkamenosa commented:
Is this released in 2.8.1. I got the same issue, and here is my config: 
```
# Credentials for your image host.
registry:
  server: "localhost:5555"

# Configure the image builder.
builder:
  # Build for the server's native architecture (aarch64 == arm64)
  arch: arm64
  # Enable a registry cache so apt/bundler layers can be reused across deploys.
  cache:
    type: registry
    # Use an arch-specific cache image to avoid mixing caches across platforms
    image: tuyenhx/async_example-buildcache-arm64
    options: mode=max,image-manifest=true,oci-mediatypes=true

  remote: ssh://root@46.224.21.86
  local: false
```

djmb commented:
@adrienpoly - I only see remote builds failing in the action tab. I think the "connection reset by peer" error probably indicates that port forwarding was initially set up correctly, so possibly the SSH connection is being dropped?

I've added the keepalive configuration, hopefully that might fix it, but I've also added some logging for any errors in the port forwarding thread, if anything goes wrong there right now I think we are missing it.

Could you try building with this branch again when you have a chance?
